src/index.ts

```diff
@@ -104,28 +104,81 @@ export interface BigtableOptions extends gax.GoogleAuthOptions {
 }
 
 /**
- * Retrieves the domain to be used for the service path.
+ * Retrieves the universe domain, if configured.
  *
- * This function retrieves the domain from gax.ClientOptions passed in or via an environment variable.
- * It defaults to 'googleapis.com' if none has been set.
- * @param {string} [prefix] The prefix for the domain.
- * @param {gax.ClientOptions} [opts] The gax client options
- * @returns {string} The universe domain.
+ * This function checks for a universe domain in the following order:
+ * 1. The `universeDomain` property within the provided options.
+ * 2. The `universeDomain` or `universe_domain` property within the `opts` object.
+ * 3. The `GOOGLE_CLOUD_UNIVERSE_DOMAIN` environment variable.
+ *
+ * If a universe domain is found in any of these locations, it is returned.
+ * Otherwise, the function returns `undefined`.
+ *
+ * @param {BigtableOptions} options - The Bigtable client options.
+ * @param {gax.ClientOptions} [gaxOpts] - Optional gax client options.
+ * @returns {string | undefined} The universe domain, or `undefined` if not found.
  */
-function getDomain(prefix: string, opts?: gax.ClientOptions) {
+function getUniverseDomainOnly(
+  options: BigtableOptions,
+  gaxOpts?: gax.ClientOptions,
+): string | undefined {
   // From https://github.com/googleapis/nodejs-bigtable/blob/589540475b0b2a055018a1cb6e475800fdd46a37/src/v2/bigtable_client.ts#L120-L128.
   // This code for universe domain was taken from the Gapic Layer.
   // It is reused here to build the service path.
   const universeDomainEnvVar =
     typeof process === 'object' && typeof process.env === 'object'
       ? process.env['GOOGLE_CLOUD_UNIVERSE_DOMAIN']
       : undefined;
-  return `${prefix}.${
-    opts?.universeDomain ??
-    opts?.universe_domain ??
-    universeDomainEnvVar ??
-    'googleapis.com'
-  }`;
+  return (
+    options?.universeDomain ??
+    gaxOpts?.universeDomain ??
+    gaxOpts?.universe_domain ??
+    universeDomainEnvVar
```

Comment:
what is the difference between `options.universeDomain` and `gaxOpts.universeDomain` from a user perspective? What does it look like when one overrides the other?

Reply:
`gaxOpts` is the settings the user provided for the specific Gapic client and `options` is the options provided to the handwritten layer. We discussed offline that we should really allow the gapic options to take priority over the handwritten layer options so I can apply the change to make gaxOpts?.universeDomain come first.

Reply:
Ok. gaxOpts?.universeDomain comes first now